lib/maropost/api.rb

```diff
@@ -0,0 +1,58 @@
+module Maropost
+  class Api
+    include ActiveModel::Model
+
+    attr_accessor :url, :http_method, :data, :template_name
+
+    def create
+      self.url = "#{Maropost.configuration.api_url}/contacts.json"
+      self.http_method = :post
+      self.template_name = "create_or_update.json.erb"
+      request
+    end
+
+    def find
+      self.url = "#{Maropost.configuration.api_url}/contacts/email.json?contact[email]=#{data[:email]}"
+      self.http_method = :get
+      self.template_name = "find.json.erb"
+      request
+    end
+
+    def update
+      self.url = "#{Maropost.configuration.api_url}/contacts/#{data[:id]}.json"
+      self.http_method = :put
+      self.template_name = "create_or_update.json.erb"
+      request
+    end
+
+    def update_email
+      self.url = "#{Maropost.configuration.api_url}/contacts/#{data[:id]}.json"
+      self.http_method = :put
+      self.template_name = "update_email.json.erb"
+      request
+    end
+
+  private
+    def request
+      RestClient::Request.execute(
```

Comment:
Can we use the `RestClient::Request.logged_request` method here so we get Rollbar and Loggly events>

https://github.com/amaabca/api/blob/6ea89eafa7af5de8535abb250d2b44b31b75e2db/app/models/bwb_rate.rb#L16